components/raftstore-v2/src/operation/ready/mod.rs

```diff
@@ -370,6 +370,17 @@ impl<EK: KvEngine, ER: RaftEngine> Peer<EK, ER> {
             // It prevents cache peer(0,0) which is sent by region split.
             self.insert_peer_cache(from_peer);
         }
+
+        // Delay first message and wait for split snapshot, so that slow split
+        // does not trigger leader to send a snapshot.
+        if !self.storage().is_initialized() && is_first_message(msg.get_message()) {
+            if self.split_pending_msg_mut().replace(msg).is_some() {
```

Comment:
Is the new msg same as the existing one in split_pending_msg_mut()?
If it's same, then we can add an assert. If it's different, then the new message may be stepped before the old message, which could lead to correctness problem.

Reply:
New msgs can be different.

It's okay to step messages out of order.